.github/workflows/build.yml

```
@@ -3,16 +3,73 @@ name: build
 on: [push, pull_request]
 
 jobs:
+  gitversion:
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.gitversion.outputs.fullSemVer }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: gittools/actions/gitversion/setup@v0.9.7
+        with:
+          versionSpec: '5.x'
+      - uses: gittools/actions/gitversion/execute@v0.9.7
+        id: gitversion
+        with:
+          useConfigFile: true
+          configFilePath: .github/gitversion.yml
+
   build:
+    needs: gitversion
     strategy:
       matrix:
         os: [macos-latest, ubuntu-latest, windows-latest]
     runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        env:
+          cache-name: cache-npm
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: ${{ runner.os }}-build-${{ env.cache-name }}-
+      - name: npm version
+        run: npm version --no-git-tag-version ${{ needs.gitversion.outputs.version }}
       - run: npm install
       - run: npm run build
-      - name: run headless test
+      - uses: actions/upload-artifact@v2
+        with:
+          name: vscode-remark
+          path: |
+            out
+            package*.json
+      - name: test
         uses: GabrielBB/xvfb-action@v1.4
         with:
           run: npm test
+
+  publish-gpr:
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/cache@v2
+        env:
+          cache-name: cache-npm
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: ${{ runner.os }}-build-${{ env.cache-name }}-
+      - uses: actions/download-artifact@v2
+        with:
+          name: vscode-remark
+      - run: npm install
+      - uses: actions/setup-node@v2
+        with:
+          registry-url: https://npm.pkg.github.com/
+      - name: npm config
+        run: echo "registry=https://npm.pkg.github.com/@remarkjs" >> .npmrc
```

```suggestion
        run: echo "registry=https://npm.pkg.github.com" >> .npmrc
```

One more thing to try.